Doc/library/io.rst

```diff
@@ -106,6 +106,54 @@ stream by opening a file in binary mode with buffering disabled::
 The raw stream API is described in detail in the docs of :class:`RawIOBase`.
 
 
+.. _io-text-encoding:
+
+Text Encoding
+-------------
+
+The default encoding of :class:`TextIOWrapper` and :func:`open` is
+locale-specific (:func:`locale.getpreferredencoding(False) <locale.getpreferredencoding>`).
+
+But many developers forget to specify encoding when opening text files
+encoded in UTF-8 (e.g. JSON, TOML, Markdown, etc...) since most Unix
+platforms uses UTF-8 locale by default. It cause a bug because locale-specific
+encoding is not UTF-8 for most Windows users. For example::
+
+   # may not work on Windows when non-ASCII characters in the file.
+   with open("README.md") as f:
+       long_description = f.read()
+
+Additionally, Python may change the default text encoding to UTF-8 in the
+future, although there is no plan yet.
+
+So it is highly recommended to specify encoding explicitly when opening text
+files. If you want to use UTF-8, specify ``encoding="utf-8"``. If you need to
+use locale-specific encoding, ``encoding="locale"`` is supported since Python
+3.10.
+
+When you need to run code using the default encoding to open UTF-8 files on
+Windows, you can enable the UTF-8 mode. See also the
+:ref:`UTF-8 mode on Windows <win-utf8-mode>`
+
+.. _io-encoding-warning:
+
+Opt-in EncodingWarning
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 3.10
+   See :pep:`597` for more details.
+
+To find where the default encoding is used, you can use
+a ``-X warn_default_encoding`` command line argument or a
+:envvar:`PYTHONWARNDEFAULTENCODING` environment variable to emit
+an :exc:`EncodingWarning` when the defaut encoding is used.
+
+If you are providing APIs using :func:`open` or :class:`TextIOWrapper` and
+having ``encoding=None`` parameter, you can use :func:`text_encoding` to emit
+an :exc:`EncodingWarning` to the user too. But please consider using UTF-8
+by default (i.e. ``encoding="utf-8"``).
```

Comment:
```suggestion
If you are providing an API that uses :func:`open` or
:class:`TextIOWrapper` and passes ``encoding=None`` as a parameter, you
can use :func:`text_encoding` to emit an :exc:`EncodingWarning` to the
user too. However, please consider using UTF-8 by default
(i.e. ``encoding="utf-8"``).
```

Fix a few grammar issues (agreement, articles, incomplete sentences)

Reply:
> "to the user too. "

This is a bit misleading. When a function wrapping `open()` uses `encoding = io.text_encoding(encoding)`, the function is not warned. Only caller of the function will be warned.

Reply:
Okay, thanks. I've tried to clarify it accordingly; is this better?

```suggestion
If you are providing an API that uses :func:`open` or
:class:`TextIOWrapper` and passes ``encoding=None`` as a parameter, you
can use :func:`text_encoding` so that callers of the API will emit an
:exc:`EncodingWarning` if they don't pass an ``encoding``. However,
please consider using UTF-8 by default (i.e. ``encoding="utf-8"``) for
new APIs.
```